sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcQuerySuite.scala

```diff
@@ -832,6 +832,18 @@ abstract class OrcQuerySuite extends OrcQueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-39387: BytesColumnVector should not throw RuntimeException due to overflow") {
```

Comment:
I saw one test failure with this in JDK 11:

```
[info] - SPARK-39387: BytesColumnVector should not throw RuntimeException due to overflow *** FAILED *** (3 seconds, 393 milliseconds)
[info]   org.apache.spark.SparkException: Job aborted.
[info]   at org.apache.spark.sql.errors.QueryExecutionErrors$.jobAbortedError(QueryExecutionErrors.scala:593)
[info]   at org.apache.spark.sql.execution.datasources.FileFormatWriter$.write(FileFormatWriter.scala:279)
[info]   at org.apache.spark.sql.execution.datasources.InsertIntoHadoopFsRelationCommand.run(InsertIntoHadoopFsRelationCommand.scala:186)
[info]   at org.apache.spark.sql.execution.command.DataWritingCommandExec.sideEffectResult$lzycompute(commands.scala:113)
[info]   at org.apache.spark.sql.execution.command.DataWritingCommandExec.sideEffectResult(commands.scala:111)
[info]   at org.apache.spark.sql.execution.command.DataWritingCommandExec.executeCollect(commands.scala:125)
[info]   at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.$anonfun$applyOrElse$1(QueryExecution.scala:98)
[info]   at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:111)
[info]   at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:171)
[info]   at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)
[info]   at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)
[info]   at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
[info]   at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:98)
[info]   at org.apache.spark.sql.execution.QueryExecution$$anonfun$eagerlyExecuteCommands$1.applyOrElse(QueryExecution.scala:94)
[info]   at org.apache.spark.sql.catalyst.trees.TreeNode.$anonfun$transformDownWithPruning$1(TreeNode.scala:584)
[info]   at org.apache.spark.sql.catalyst.trees.CurrentOrigin$.withOrigin(TreeNode.scala:176)
[info]   at org.apache.spark.sql.catalyst.trees.TreeNode.transformDownWithPruning(TreeNode.scala:584)
```

https://github.com/apache/spark/runs/6919076419?check_suite_focus=true

I filed a JIRA at SPARK-39515. Would be great if we can identify if it was just flakiness or actual test failure with JDK 11.

Reply:
Thanks for reporting the problem, I'll look into it.

Reply:
I tested it with JDK11 locally and it can run successfully.
```bash
setjdk 1.11
build/mvn -Dtest=none -DwildcardSuites=org.apache.spark.sql.execution.datasources.orc.OrcV1QuerySuite test
```
![image](https://user-images.githubusercontent.com/3898450/174632802-10abcf43-d1df-4b1b-a8ac-097f240338d6.png)



I saw the GA error because the writing process encountered OOM, which should have nothing to do with JDK11.
```java
2022-06-16T14:30:19.8285352Z Caused by: java.lang.OutOfMemoryError: Java heap space
2022-06-16T14:30:19.8285963Z 	at org.apache.hadoop.hive.ql.exec.vector.BytesColumnVector.allocateBuffer(BytesColumnVector.java:300)
2022-06-16T14:30:19.8286885Z 	at org.apache.hadoop.hive.ql.exec.vector.BytesColumnVector.ensureValPreallocated(BytesColumnVector.java:218)
2022-06-16T14:30:19.8287675Z 	at org.apache.hadoop.hive.ql.exec.vector.BytesColumnVector.setVal(BytesColumnVector.java:182)
2022-06-16T14:30:19.8288377Z 	at org.apache.orc.mapred.OrcMapredRecordWriter.setBinaryValue(OrcMapredRecordWriter.java:87)
2022-06-16T14:30:19.8289257Z 	at org.apache.orc.mapred.OrcMapredRecordWriter.setColumn(OrcMapredRecordWriter.java:235)
2022-06-16T14:30:19.8289956Z 	at org.apache.orc.mapred.OrcMapredRecordWriter.setStructValue(OrcMapredRecordWriter.java:133)
2022-06-16T14:30:19.8290654Z 	at org.apache.orc.mapred.OrcMapredRecordWriter.setColumn(OrcMapredRecordWriter.java:248)
2022-06-16T14:30:19.8291438Z 	at org.apache.orc.mapred.OrcMapredRecordWriter.setListValue(OrcMapredRecordWriter.java:162)
2022-06-16T14:30:19.8292127Z 	at org.apache.orc.mapred.OrcMapredRecordWriter.setColumn(OrcMapredRecordWriter.java:256)
2022-06-16T14:30:19.8292824Z 	at org.apache.orc.mapreduce.OrcMapreduceRecordWriter.write(OrcMapreduceRecordWriter.java:73)
2022-06-16T14:30:19.8293554Z 	at org.apache.spark.sql.execution.datasources.orc.OrcOutputWriter.write(OrcOutputWriter.scala:56)
2022-06-16T14:30:19.8294523Z 	at org.apache.spark.sql.execution.datasources.SingleDirectoryDataWriter.write(FileFormatDataWriter.scala:175)
```

This test does not seem to be able to compress buffer memory like PR #34284, it requires a relatively large memory to write to ORC to ensure test coverage.

Reply:
Thanks, man!

Reply:
> ```shell
> ```shell
> build/mvn -Dtest=none -DwildcardSuites=org.apache.spark.sql.execution.datasources.orc.OrcV1QuerySuite test
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> ```


When mvn test the whole `sql/core` module, it is easy to reproduce OOM on this case, even with Java 8. I think we should consider simplifying the case or increasing the test memory

Reply:
Thanks for the feedback, I'll see how to optimize its memory usage.

Reply:
@cxzl25 In https://github.com/apache/spark/pull/36954, I am trying to solve the issue by changing JVM options. If you have solved this issue in other ways , please tell me ~ thanks ~

Reply:
This UT requires a large heap, using end-to-end writing orc, and covering tests.

Maybe we can use ignore first.

Based on this comment https://github.com/apache/spark/pull/36772#issuecomment-1147811164, Hive also increased `xmx` to make UT pass.

I tried to reduce the size of each batch write, but because there are still some temporary buffers in the process of spark to orc writing, it may oom.
And if the oom error is intercepted, the Spark process will exit abnormally, and the ut will fail.